framework/src/meshgenerators/OverlayMeshGenerator.C

```diff
@@ -0,0 +1,138 @@
+      1.,
+      "bias_x>=0.5 & bias_x<=2",
+      "The amount by which to grow (or shrink) the cells in the x-direction.");
+  params.addRangeCheckedParam<Real>(
+      "bias_y",
+      1.,
+      "bias_y>=0.5 & bias_y<=2",
+      "The amount by which to grow (or shrink) the cells in the y-direction.");
+  params.addRangeCheckedParam<Real>(
+      "bias_z",
+      1.,
+      "bias_z>=0.5 & bias_z<=2",
+      "The amount by which to grow (or shrink) the cells in the z-direction.");
+  return params;
+}
+
+OverlayMeshGenerator::OverlayMeshGenerator(const InputParameters & parameters)
+  : MeshGenerator(parameters),
+    _dim(getParam<MooseEnum>("dim")),
+    _mesh_name(getParam<MeshGeneratorName>("input"))
+{
+  // Declare that all of the meshes in the "inputs" parameter are to be used by
+  // a sub mesh generator
+  declareMeshForSub("input");
+
+  _input_mesh = &getMeshByName(_mesh_name);
+
+  auto params = _app.getFactory().getValidParams("DistributedRectilinearMeshGenerator");
+
+  params.set<MooseEnum>("dim") = _dim;
+
+  params.set<dof_id_type>("nx") = getParam<dof_id_type>("nx");
+  params.set<dof_id_type>("ny") = getParam<dof_id_type>("ny");
+  params.set<dof_id_type>("nz") = getParam<dof_id_type>("nz");
+
+  params.set<Real>("xmin") = 0;
+  params.set<Real>("ymin") = 0;
+  params.set<Real>("zmin") = 0;
+  params.set<Real>("xmax") = 1;
+  params.set<Real>("ymax") = 1;
+  params.set<Real>("zmax") = 1;
+
+  params.set<Real>("bias_x") = getParam<Real>("bias_x");
+  params.set<Real>("bias_y") = getParam<Real>("bias_y");
+  params.set<Real>("bias_z") = getParam<Real>("bias_z");
+
+  params.set<unsigned>("num_side_layers") = getParam<unsigned>("num_side_layers");
+  params.set<processor_id_type>("num_cores_for_partition") =
+      getParam<processor_id_type>("num_cores_for_partition");
+
+  params.set<MooseEnum>("partition") = getParam<MooseEnum>("partition");
+  params.set<MooseEnum>("elem_type") = getParam<MooseEnum>("elem_type");
+
+  addMeshSubgenerator("DistributedRectilinearMeshGenerator",
+                      _mesh_name + "_distributedrectilinearmeshgenerator",
+                      params);
+  _build_mesh = &getMeshByName(_mesh_name + "_distributedrectilinearmeshgenerator");
+}
+std::unique_ptr<MeshBase>
+OverlayMeshGenerator::generate()
+{
+  std::unique_ptr<MeshBase> input_mesh = std::move(*_input_mesh);
+  std::unique_ptr<MeshBase> build_mesh = std::move(*_build_mesh);
+
+  // find the boundary of the input mesh box
+  auto bbox_input = MeshTools::create_bounding_box(*input_mesh);
+
+  // Transform the generated DistributedRectilinearMesh to overlay with the input mesh
+  RealVectorValue scale_factor;
+  scale_factor = bbox_input.max() - bbox_input.min();
+
+  // scale
+  if (scale_factor(0) != 1 || scale_factor(1) != 1 || scale_factor(2) != 1)
+    MeshTools::Modification::scale(*build_mesh, scale_factor(0), scale_factor(1), scale_factor(2));
+
+  RealVectorValue translate_factor;
+  translate_factor = bbox_input.min();
```

Comment:
```suggestion
  translation_vector = bbox_input.min();
```

factor is only for multiplication